### DISCUSSION TITLE
Disabled a field in side panel based on the value in it

### DISCUSSION THREAD

**Discussion by Mozix** [Budibase/budibase#12370] [Help] [answered]
Is it possible to disable specific fields in a side panel form based on what's inside the field? so the user shouldn't be able to edit fields that have something inside but can edit those that have nothing inside.

**Comment by sdsys-ch** [comment, marked as answer]
Hi @Mozix 

You can set a condition for each form field after ejecting the specific block(s) to its base elements. Then set two conditions for the `Disabled` setting via `Update setting` in `Conditions`:

![image](https://github.com/Budibase/budibase/assets/100968265/8bdcfaca-3843-4e3f-8224-2bad1833a09b)

Do this for every relevant form field and it works as expected:

https://github.com/Budibase/budibase/assets/100968265/fc87f531-6475-435f-a928-4e20493e497a

**Comment by Mozix** [comment]
Okay would it be doable using a New Form Block and not a custom one?

**Comment by sdsys-ch** [comment]
I didn't build mine individually. I asked budibase for a new screen with a pre-built table linked to an existing db table, ejected the (generated) main-window Table Block and then the side-panel block. That's where I then added the conditionals.

Ejecting the `Table block`:

<img width="388" alt="image" src="https://github.com/Budibase/budibase/assets/100968265/02712821-0fb4-4a1f-b6cc-e8190bc9fb60">

becomes 

<img width="377" alt="image" src="https://github.com/Budibase/budibase/assets/100968265/c9505aef-e79d-40a6-9246-97894689c09c">

where you then eject the `Details form block`:

<img width="302" alt="image" src="https://github.com/Budibase/budibase/assets/100968265/76b0ab62-14db-4c01-b886-d4eb18a4083e">

where you can then add the conditionals to the `Fieldgroup`'s individual fields. You even get to keep the ability to update the `Formgroup`'s fields after a schema change, albeit loosing the field conditionals configuration in the process.

Unless I didn't get what you mean?

I dont' see an easy way to toggle the `Disable`d state for individual Form fields that have no values when keeping the Form block intact, but that might also be due to my so far limited budibase experience. Someone else might be able to chime in on this. Maybe using a form's conditional for `Fields` paired with some JS or the like, but that seems overly complicated.

**Reply by Mozix** [reply]
Ejecting blocking fixed the issue! I did not know that was a thing hahaha but it gave me more control over what I need!

Thank you so much!

**Reply by sdsys-ch** [reply]
Glad this helped and yes, I also had that _aaaaah_ moment when I found out

**Reply by Mozix** [reply]
Hahahah i feel you, the moment I saw these components split and it was all there i just gasped and had my mouth wide open going like oh my goooooooood
